### PR TITLE
Feature/transaction status cli

### PR DESCRIPTION
**Pull request by wu-s-john**
Transaction status check via CLI

### CODE REVIEW

File: src/lib/transaction_status/dune [hunk 1]
```diff
@@ -2,4 +2,4 @@
   (name transaction_status)
   (public_name transaction_status)
   (libraries core_kernel coda_base transition_frontier network_pool)
-  (preprocess (pps ppx_jane)))
\ No newline at end of file
+  (preprocess (pps ppx_jane ppx_coda)))
```
Comment (figitaki) [inline src/lib/transaction_status/dune]: 👀

File: src/lib/base58_check/version_bytes.ml [hunk 2]
```diff
@@ -4,7 +4,7 @@ type t = char
 
 (* each of the following values should be distinct *)
 
-let graphql : t = '\x17'
+let user_command : t = '\x17'
```
Comment (figitaki) [inline src/lib/base58_check/version_bytes.ml]: we should def document this somewhere

File: src/app/cli/src/client.ml [hunk 3]
```diff
@@ -467,6 +470,26 @@ let send_payment =
...
+               ~success:
+                 (Or_error.map ~f:(fun status ->
+                      sprintf !"Transaction status:%s\n"
+                      @@ Transaction_status.State.to_string status ))
+               ~error:(fun e ->
+                 sprintf "Failed to get transaction status: %s"
+                   (Error.to_string_hum e) )
+         | Error _e ->
+             eprintf "Could not deserialize user command" ;
+             exit 1 ))
```
Comment (cmr) [inline src/app/cli/src/client.ml]: We're keeping exit codes unique (and, specifically, non-1) for now to help with the "mysterious exit 1".  It looks like the next available exit code is 16.

File: src/app/cli/src/client.ml [hunk 4]
```diff
@@ -467,6 +470,26 @@ let send_payment =
...
+    (Cli_lib.Background_daemon.init
+       Command.Param.(anon @@ ("txn" %: string))
+       ~f:(fun port serialized_transaction ->
+         match User_command.of_base58_check serialized_transaction with
+         | Ok user_command ->
+             dispatch_with_message Daemon_rpcs.Get_transaction_status.rpc
+               user_command port
+               ~success:
+                 (Or_error.map ~f:(fun status ->
+                      sprintf !"Transaction status:%s\n"
```
Comment (bkase) [inline src/app/cli/src/client.ml]: nit: put a space after the `:` here

File: src/app/cli/src/client.ml [hunk 5]
```diff
@@ -440,14 +440,17 @@ let user_command (body_args : User_command_payload.Body.t Command.Param.t)
...
+         let payment =
+           (User_command.sign sender_kp payload :> User_command.t)
+         in
+         dispatch_with_message Daemon_rpcs.Send_user_command.rpc payment port
            ~success:
              (Or_error.map ~f:(fun receipt_chain_hash ->
-                  sprintf "Initiated %s\nReceipt chain hash: %s" label
-                    (Receipt.Chain_hash.to_string receipt_chain_hash) ))
+                  sprintf
+                    "Initiated %s\nReceipt chain hash: %s\nTransaction ID:%s\n"
```
Comment (bkase) [inline src/app/cli/src/client.ml]: Can you change this to say:

```
sprintf
"Dispatched %s with ID %s\nReceipt chain hash is now %s\n"
                    label
                    (Receipt.Chain_hash.to_string receipt_chain_hash)
                    (User_command.to_base58_check payment)
```

**Comment by enolan** [comment]
Will take a look when I get into the office.

File: src/app/cli/src/client.ml [hunk 6]
```diff
@@ -440,14 +440,17 @@ let user_command (body_args : User_command_payload.Body.t Command.Param.t)
...
+         in
+         dispatch_with_message Daemon_rpcs.Send_user_command.rpc payment port
            ~success:
              (Or_error.map ~f:(fun receipt_chain_hash ->
-                  sprintf "Initiated %s\nReceipt chain hash: %s" label
-                    (Receipt.Chain_hash.to_string receipt_chain_hash) ))
+                  sprintf
+                    "Dispatched %s with ID %s\nReceipt chain hash is now %s\n"
+                    label
+                    (Receipt.Chain_hash.to_string receipt_chain_hash)
```
Comment (enolan) [inline src/app/cli/src/client.ml]: You have the receipt chain hash and the base58check payment swapped here. They should switch places.

File: src/app/cli/src/serialization.ml [hunk 7]
```diff
@@ -0,0 +1,21 @@
+open Core
```
Comment (enolan) [inline src/app/cli/src/serialization.ml]: I don't think this module is used anywhere. ctrl-f `serialization` doesn't return any relevant results.

File: src/lib/base58_check/version_bytes.ml (hunk 2 above)
Comment (psteckler) [reply src/lib/base58_check/version_bytes.ml]: document the particular version bytes used?

**Pull request merged**